### ISSUE TITLE
VS Code Extension Feature Ideas

### ISSUE THREAD

**Issue by jonah-iden** [TypeFox/open-collaboration-server#2]
- [ ] Kicking User from a Session as Host
  - this would probably take some server changes since i don't think that functionality is there yet
- [ ] Jump to other user cursor location (single jump without following)
- [ ] Terminal Sharing (behind propesed api, so maybe note possible)
- [ ] Collaborative debugging (behind propesed api, so maybe note possible)
- [ ] Show open editors of other users in tree ui

**Comment by dhuebner** [comment]
I had also an idea: Show other user opened editors in "Explorer" -> "Opened Editors" tree. Show with greyed out font or other visual indicator.

**Comment by bilogic** [comment]
How about an internal browser for preview/debug? But it probably requires sending images.

**Comment by bilogic** [comment]
And preview of SVG/PNGs etc? I tried on collaborator side, these files actually loads if collaborator clicks on them, but currently don't show up if sharer is focus on them.

**Comment by msujew** [comment]
> And preview of SVG/PNGs etc? I tried on collaborator side, these files actually loads if collaborator clicks on them, but currently don't show up if sharer is focus on them.

So you're talking about the follower mode?

> How about an internal browser for preview/debug? But it probably requires sending images instead of loading the same URL.

I've already thought about a tunnel mode for previewing URLs, but it's quite a bit of effort to get this right. It's a good idea to put this on the roadmap though.

**Comment by bilogic** [comment]
> > And preview of SVG/PNGs etc? I tried on collaborator side, these files actually loads if collaborator clicks on them, but currently don't show up if sharer is focus on them.
> 
> So you're talking about the follower mode?

Yes, (C)ollaborator can open the same file if he clicks on his end, it's just that it currently does not open by itself if the (S)harer opens it.

> I've already thought about a tunnel mode for previewing URLs, but it's quite a bit of effort to get this right. It's a good idea to put this on the roadmap though.

Actually, I would like to help triage the list of to dos, in short, I think the highest impact item is terminal sharing because without a browser, we can still unit test. That completes the full development cycle.

**Comment by bilogic** [comment]
## Triage

1. Terminal Sharing
   - big impact: all can see the outcome of running tests/modified code (since browser is much hard to share)

2. Collaborative debugging 
   - big impact: getting a more experience developer to assist

3. Show the same open editors
   - Impact: puts the developers on the same viewing experience, as we might have 2-3 columns to view 2-3 files together

4. Have a list of users on the side bar, clicking on a user, we will be able to join immediately as a collaborator
   - Impact: useful for teachers jumping around to see what their students are doing

5. Focus on user (single jump user current location)

**Comment by HKalbasi** [comment]
+1 for terminal sharing. This is the number one thing I miss from my vim+zellij workflow.
